libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -225,42 +215,67 @@ private <T> T getJsonProperty(
                                 .getTypeFactory()
-                                .constructCollectionType(List.class, Name.class),
-                        validateCorrelation);
+                                .constructCollectionType(List.class, Name.class));
         List<BirthDate> birthDates =
                 getJsonProperty(
                         vcClaimNode,
                         BIRTH_DATE_PROPERTY_NAME,
-                        credentialIssuer,
                         objectMapper
                                 .getTypeFactory()
-                                .constructCollectionType(List.class, BirthDate.class),
-                        validateCorrelation);
+                                .constructCollectionType(List.class, BirthDate.class));
 
         return new IdentityClaim(names, birthDates);
     }
 
-    private Optional<IdentityClaim> generateIdentityClaim(List<VcStoreItem> successfulVCStoreItems)
+    public Optional<IdentityClaim> generateIdentityClaim(List<VcStoreItem> successfulVCStoreItems)
             throws HttpResponseExceptionWithErrorBody {
-        for (VcStoreItem item : successfulVCStoreItems) {
-            if (EVIDENCE_CRI_TYPES.contains(item.getCredentialIssuer())) {
-                return Optional.of(
-                        getIdentityClaim(item.getCredential(), item.getCredentialIssuer(), false));
+        if (successfulVCStoreItems.isEmpty()) {
+            LOGGER.warn("Failed to generate identity claim");
+            return Optional.empty();
+        }
+        IdentityClaim identityClaim = findIdentityClaim(successfulVCStoreItems);
+        return Optional.of(identityClaim);
+    }
+
+    public IdentityClaim findIdentityClaim(List<VcStoreItem> vcStoreItems)
+            throws HttpResponseExceptionWithErrorBody {
+        List<IdentityClaim> identityClaims = new ArrayList<>();
+        for (VcStoreItem item : vcStoreItems) {
+            try {
+                if (isEvidenceVc(item)) {
+                    identityClaims.add(getIdentityClaim(item.getCredential()));
+                }
+            } catch (CredentialParseException e) {
+                LOGGER.error("Failed to parse VC JWT because: {}", e.getMessage());
+                throw new HttpResponseExceptionWithErrorBody(
+                        500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
             }
         }
-        LOGGER.warn("Failed to generate identity claim");
-        return Optional.empty();
+
+        Optional<IdentityClaim> claimWithName =
+                identityClaims.stream()
+                        .filter(identityClaim -> !identityClaim.getName().isEmpty())
+                        .findFirst();
+        Optional<IdentityClaim> claimWithBirthDate =
+                identityClaims.stream()
+                        .filter(identityClaim -> !identityClaim.getBirthDate().isEmpty())
+                        .findFirst();
+        if (claimWithName.isEmpty() || claimWithBirthDate.isEmpty()) {
```

Comment:
Is there a CRI type that we should exclude before throwing an errors ere? For example BAV, Address CRI or, are we expecting at least one of the successful VCs to have a name and a birthdate?

Reply:
My assumption for this ticket was that if we can't find a name & date of birth on the evidence VCs we should throw an error here. Do we need to handle the case where the `identityClaims` list is empty and return empty, similar to here [return Optional.of(identityClaim);](https://github.com/govuk-one-login/ipv-core-back/pull/1292/files/705778f645f1f3b4a101cfbbbebd58b3f8d055e6#diff-03e309f6bbe7b1e939e8d7a3f5ad31be61b375cf9c88a8a05048bed7970df470R246)

Reply:
Yeah, you are right. According to the task requirements, it appears that both name and dob are expected to be present in at least one successful VC, In Name and DOB correlations, I am excluding the following 

- DOB correlation excludes BAV and Address CRI
- Name correlation excludes only Address CRI

Because, Address CRI does not a name and DOB, BAV CRI does not a DOB.